net-utils/src/lib.rs

```diff
@@ -384,29 +386,145 @@ pub fn is_host_port(string: String) -> Result<(), String> {
     parse_host_port(&string).map(|_| ())
 }
 
+lazy_static! {
```

Comment:
Are we not doing the same thing that the OS would do for us here? We can always figure out which ports are used by just asking the OS. And it would be far better since we are not the only process using ports.